### PR TITLE
feat: implement OS notifications on Linux

### PR DESCRIPTION
**Pull request by caybro**
using "notify-send" wrapper

Related: status-im/status-desktop#2520

### CODE REVIEW

File: lib/src/Status/OSNotification.cpp [hunk 1]
```diff
@@ -158,6 +164,22 @@ void OSNotification::showNotification(const QString& title,
...
 #elif defined Q_OS_MACOS
     showNotificationMacOs(title, message, identifier);
+#elif defined Q_OS_LINUX
+    static QString notifyExe = QStandardPaths::findExecutable(QStringLiteral("notify-send"));
+    if (notifyExe.isEmpty()) {
+        qWarning() << "'notify-send' not found; OS notifications will not work";
+        return;
+    }
+
+    //qDebug() << "NOTIFY-SEND; title:" << title << "; message:" << message << "; id:" << identifier;
```
Comment (saledjenic) [inline lib/src/Status/OSNotification.cpp:174]: @caybro is this needed?

**Review by saledjenic — APPROVED** [review]
@caybro as I see this is for adding/displaying notifications, do we have a way to handle a user's click on a certain OS notification?

File: lib/src/Status/OSNotification.cpp (hunk 1 above)
Comment (caybro) [reply lib/src/Status/OSNotification.cpp:174]: Nope, I can drop that, was just used when debugging this

**Comment by caybro** [comment]
> @caybro as I see this is for adding/displaying notifications, do we have a way to handle a user's click on a certain OS notification?

Unfortunately not, this is a very rudimentary "backend" usign the cmd line "notify-send" tool. In order to fully support clicking the OS notification (and other things, like actions, etc), I'd need to use DBUS (maybe later :)

File: lib/src/Status/OSNotification.cpp (hunk 1 above)
Comment (caybro) [reply lib/src/Status/OSNotification.cpp]: Done

File: lib/src/Status/OSNotification.cpp [hunk 2]
```diff
@@ -158,6 +164,20 @@ void OSNotification::showNotification(const QString& title,
 
 #elif defined Q_OS_MACOS
     showNotificationMacOs(title, message, identifier);
+#elif defined Q_OS_LINUX
+    static QString notifyExe = QStandardPaths::findExecutable(QStringLiteral("notify-send"));
```
Comment (osmaczko) [inline lib/src/Status/OSNotification.cpp:168]:
```suggestion
    static const QString notifyExe = QStandardPaths::findExecutable(QStringLiteral("notify-send"));
```

**Review by osmaczko — APPROVED** [review]
LGTM

**Pull request merged**